### PR TITLE
Added development environment guide to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by PeterFogh**
Hi, it's my first time contributing to eo-learn, thus I think it was fitting to improve the contribution guide.
I have contributed to other open-source before, e.g. DVC and think their contribution guide (https://dvc.org/doc/user-guide/contributing) has some details the eo-learn guide is missing, for example, the guide of setting up a virtual Python environment for development which I have documented in this pull request.

A guide for executing the unittests could also be useful. However, I'm not familiar enough with your test setup to document such a guide. Are somebody else up for that?

### CODE REVIEW

**Comment by PeterFogh** [comment]
Additionally, I can add that I use Conda for my development environment, by the following installed:
````
conda create -n open_source_eolearn python=3.7
conda activate open_source_eolearn
conda install ipykernel # enables Jupyter to select the kernel
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/core
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/coregistration
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/features
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/geometry
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/io
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/mask
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/ml_tools
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn/visualization
pip install -e /home/ubuntu_FOGH/work/open_source/eo-learn
```

**Comment by mlubej** [comment]
Hi Peter!

Thanks a lot for your contribution and help. Adding the `conda` guide would also be helpful, I don't think that we as a team have a specific preference towards anything, it's just that there were more problems with `conda` in the past, but if you managed to find a working combination, that's great to hear!

We will take a look and merge as soon as possible, but at the moment we're a bit busy with other projects.

Thanks a lot!
Cheers,
Matic

**Comment by PeterFogh** [comment]
Hi, @mlubej 

> Adding the `conda` guide would also be helpful

Super. Then, I will add a description of my Conda environment setup.

**Comment by PeterFogh** [comment]
@mlubej and others, the final improved CONTRIBUTING.md file is ready for review. :)

**Comment by PeterFogh** [comment]
I'm not sure if requirements-docs.txt should be part of the developer environment or if contributions to the documentation should be done using a separate environment?

Currently, I have not tried to build the Sphinx documentation, but I have added the eo-learn subpackages in requirements-docs.txt to be installed in developer mode (i.e. `-e` flagged) in the developer environment. If this is a problem for building the documentation, either locally or on Travis, please explain why 😄

**Comment by mlubej** [comment]
Hi @PeterFogh!

Thanks a lot for adding more work into this PR. At the moment I still cannot review things, but I would like to ask you to also add some information about the existing dependencies in the `conda` environment, i.e. the combination of packages you have installed that produce a working environment.

Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,38 +64,66 @@ The following guidelines should be observed when creating a PR.
...
+
+Or alternatively by using Conda:
+
+```bash
+cd eo-learn
+# The following creates the virtual environment named "dev_eolearn" where the
+# Conda installation in located. ipykernel enables Jupyter (using
+# nb_conda_kernels) to select the environment as a kernel. ipywidgets enables
+# Jupyter to show the progress bar of EOExecutor without raising an error.
+conda create -n dev_eolearn python=3.7 ipykernel ipywidgets
```
Comment (PeterFogh) [inline CONTRIBUTING.md]: @mlubej. I'm not sure I understand what you wish. Is it these installed Conda packages: `ipykernel` and `ipywidgets`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -74,38 +64,66 @@ The following guidelines should be observed when creating a PR.
...
+
+### Development environment
+
+* Get the latest development version by creating a fork and clone the repo:
+
+```bash
+git clone git@github.com:<username>/eo-learn.git
+```
+
+* Make sure that you have python 3 installed, i.e. version 3.5 or higher. See https://www.python.org/downloads/ for general Python versions or https://www.anaconda.com/distribution/ for using Conda.
```
Comment (devisperessutti) [inline CONTRIBUTING.md]: Suggested changes:

Make sure that you have Python ... See [Python downloads][python] for general Python versions or [Anaconda distributions][conda] for using Conda.

and adding at the bottom to the rest of the links
[python]: https://www.python.org/downloads/
[conda]: https://www.anaconda.com/distribution/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -74,38 +64,66 @@ The following guidelines should be observed when creating a PR.
...
+
+* Install **eo-learn** and all subpackages in (editable) development mode with `pip install -e <package_folder>`. We strongly recommend initializing a virtual environment before installing the required packages. For example by using Python Package Index (PyPI) and virtualenv:
```
Comment (devisperessutti) [inline CONTRIBUTING.md]: all packages can be installed at once using `python install_all.py -e`. To install each package separately, run `pip install -e <package_folder>`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -74,38 +64,66 @@ The following guidelines should be observed when creating a PR.
...
+
+```bash
+cd eo-learn
+# The following creates the virtual environment in the ".env" folder.
+virtualenv --python python3 .env
+source .env/bin/activate
+
+# The following installs all eo-learn subpackages and development packages
+# using PyPI in the activated virtualenv environment.
+pip install -e ./core -e ./coregistration -e ./features -e ./geometry -e ./io \
```
Comment (devisperessutti) [inline CONTRIBUTING.md]: `python install_all.py -e`

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -74,38 +64,66 @@ The following guidelines should be observed when creating a PR.
...
+# The following creates the virtual environment named "dev_eolearn" where the
+# Conda installation in located. ipykernel enables Jupyter (using
+# nb_conda_kernels) to select the environment as a kernel. ipywidgets enables
+# Jupyter to show the progress bar of EOExecutor without raising an error.
+conda create -n dev_eolearn python=3.7 ipykernel ipywidgets
+conda activate dev_eolearn
+
+# The following installs all eo-learn subpackages and development packages
+# using PyPI in the activated Conda environment.
+pip install -e ./core -e ./coregistration -e ./features -e ./geometry -e ./io \
```
Comment (devisperessutti) [inline CONTRIBUTING.md]: `python install_all.py -e`

**Comment by PeterFogh** [comment]
@devisperessutti, thanks for commenting the PR. I have added your comments, and I think it is ready for merge. What do you think?

**Comment by devisperessutti** [comment]
Thank you @PeterFogh for this contribution!

**Pull request merged**